tracee-ebpf/tracee/containers.go

```diff
@@ -0,0 +1,240 @@
+package tracee
+
+import (
+	"bufio"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
+)
+
+// Helper functions
+
+func InsertSortUniq(d []string, s string) []string {
+
+	p := sort.SearchStrings(d, s)
+	if p < len(d) { // exists
+		return d
+	}
+	d = append(d, "")
+	copy(d[p+1:], d[p:]) // copy to search index
+	d[p] = s
+
+	return d
+}
+
+func DeleteSortUniq(d []string, s string) []string {
+
+	p := sort.SearchStrings(d, s)
+	if p == len(d) { // does not exist
+		return d
+	}
+	d = append(d[:p], d[:p+1]...)
+
+	return d
+}
+
+// Information about running containers
+
+type Containers struct {
+	cgroupv1   bool
+	cgroupv1mp string
+	cgroupv2   bool
+	cgroupv2mp string
+	mapContIds map[string][]string
+}
+
+// Add, Remove, Check, List & Get
+
+func (c *Containers) _AddContainer(contid string) {
+	if c.mapContIds == nil {
+		c.mapContIds = make(map[string][]string, 1)
+	}
+	_, ok := c.mapContIds[contid]
+	if !ok {
+		c.mapContIds[contid] = []string{}
+	}
+}
+
+func (c *Containers) _DelContainer(contid string) {
+	_, ok := c.mapContIds[contid]
+	if ok {
+		delete(c.mapContIds, contid)
+	}
+}
+
+func (c *Containers) _CheckContainer(contid string) bool {
+	_, ok := c.mapContIds[contid]
+	if !ok {
+		return false
+	}
+
+	return true
+}
+
+func (c *Containers) GetContainers() []string {
+	var conts []string
+	for k := range c.mapContIds {
+		conts = append(conts, k)
+	}
+	if len(conts) == 0 {
+		return nil
+	}
+
+	return conts
+}
+
+func (c *Containers) _AddPid(contid string, pid string) {
+	c._AddContainer(contid)
+	c.mapContIds[contid] = InsertSortUniq(c.mapContIds[contid], pid)
+}
+
+func (c *Containers) _DelPid(contid string, pid string) {
+	if c._CheckContainer(contid) == true {
+		c.mapContIds[contid] = DeleteSortUniq(c.mapContIds[contid], pid)
+	}
+}
+
+func (c *Containers) GetPids(contid string) []string {
+	if c._CheckContainer(contid) == false {
+		return nil
+	}
+	return c.mapContIds[contid]
+}
+
+// Populate info
+
+func (c *Containers) Populate() error {
+	var err error
+
+	err = c._ProcMountsCgroups()
+	if err != nil {
+		return err
+	}
+	err = c._Populate()
+	if err != nil {
```

Comment:
No need to check nil, if it is returning err anyway below. 
it can be `return err`